Source/Core/VideoCommon/RenderBase.cpp

```diff
@@ -643,6 +643,9 @@ float Renderer::CalculateDrawAspectRatio() const
   if (aspect_mode == AspectMode::Stretch)
     return (static_cast<float>(m_backbuffer_width) / static_cast<float>(m_backbuffer_height));
 
+  if (aspect_mode == AspectMode::Internal || g_ActiveConfig.bIntegerScaling)
+    return (static_cast<float>(m_last_xfb_width) / static_cast<float>(m_last_xfb_height));
```

Comment:
Maybe I'm misunderstanding the point of the Integer Scaling feature, but why are you disabling the aspect ratio calculation when it's enabled? The point of it is that you want the output to be an exact integer multiple of the render resolution, right? If so it should be plausible to want a 4:3 or 16:9 aspect ratio but still only a multiple of the output -- you still scale on one axis in this case, of course, but the other one will be guaranteed unscaled.